scripts/add_electricity.py

```diff
@@ -318,45 +348,55 @@ def attach_storageunits(n, costs, technologies, extendable_carriers):
     lookup_store = {"H2": "electrolysis", "battery": "battery inverter"}
     lookup_dispatch = {"H2": "fuel cell", "battery": "battery inverter"}
 
-    buses_i = n.buses.index
+    number_microgrids = len(number_microgrids.keys())
+    microgrid_ids = [f"microgrid_{i+1}" for i in range(number_microgrids)]
 
-    # Iterate through each storage technology
-    for tech in technologies:
-        # Add the storage units to the power network
-        n.madd(
-            "StorageUnit",
-            buses_i,
-            " " + tech,
-            bus=buses_i,
-            carrier=tech,
-            p_nom_extendable=True,
-            capital_cost=costs.at[tech, "capital_cost"],
-            marginal_cost=costs.at[tech, "marginal_cost"],
-            efficiency_store=costs.at[
-                lookup_store["battery"], "efficiency"
-            ],  # Lead_acid and lithium have the same value
-            efficiency_dispatch=costs.at[
-                lookup_dispatch["battery"], "efficiency"
-            ],  # Lead_acid and lithium have the same value
-            max_hours=max_hours["battery"],  # Lead_acid and lithium have the same value
-            cyclic_state_of_charge=True,
-        )
+    # Iterate through each microgrid
+    for microgrid in microgrid_ids:
+        # Iterate through each storage technology
+        for tech in technologies:
+            # Add the storage units to the power network
+            n.madd(
+                "StorageUnit",
+                {microgrid},
```

Comment:
when you use "{obj}", this is a set, not a list.
Moreover, you don't need the for loop to add by microgrid. you can do something like n.madd("StorageUnit", microgrid_list, ....